storm-core/src/clj/backtype/storm/util.clj

```diff
@@ -1062,6 +1062,22 @@
 (defn hashmap-to-persistent [^HashMap m]
   (zipmap (.keySet m) (.values m)))
 
+(defn retry-on-exception
+  "Retries specific function on exception based on retries count"
+  [tries task-description f & args]
+  (let [res (try {:value (apply f args)}
+              (catch Exception e
+                (if (= 0 tries)
+                  (throw e)
+                  {:exception e})))]
+    (if (:exception res)
+      (do 
+        (log-error (:exception res) (str "Failed to " task-description ". Will make [" tries "] more attempts."))
+        (recur (dec tries) task-description f args))
+      (do 
+        (log-debug (str "Successful " task-description "."))
+        (:value res)))))
```

Comment:
Let's name the var `retries` instead of `tries`, since it calls the function `tries + 1` times.